substrate/utils/frame/remote-externalities/src/lib.rs

```diff
@@ -553,7 +574,7 @@ where
 		let mut rest = Self::get_storage_data_dynamic_batch_size(
```

Comment:
Also not sure why you have written it recursive instead of iterative. I think you should rewrite this iterative. Especially to return an error if it fails to fetch data X times in a row.